community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacade.java

```diff
@@ -193,10 +195,10 @@ public void init( SPI spi )
 
         this.spi = spi;
         this.relActions = new StandardRelationshipActions( spi::currentStatement, spi::currentTransaction,
-                this::assertTransactionOpen, ( id ) -> new NodeProxy( nodeActions, id ), this );
+                spi::assertInUnterminatedTransaction, ( id ) -> new NodeProxy( nodeActions, id ), this );
         this.nodeActions =
-                new StandardNodeActions( spi::currentStatement, spi::currentTransaction, this::assertTransactionOpen,
-                        relActions, this );
+                new StandardNodeActions( spi::currentStatement, spi::currentTransaction,
+                        spi::assertInUnterminatedTransaction, relActions, this );
```

Comment:
I wonder if we can't collect these functions in an interface at this point. Or just hand these guys the SPI reference.